### PR TITLE
Go wrapper for rd_kafka_sasl_set_credentials

### PR DESCRIPTION
**Pull request by josvisser66**
This PR adds a Go wrapper for the rd_kafka_sasl_set_credentials call in librdkafka. Not much to it, since all of the hard work (locking etc) is done in the called C function.

Note: At the time of writing the underlying function is being added to librdkafka and might appear there in v1.9.4. This PR should only be merged after the new librdkafka version containing rd_kafka_sasl_set_credentials has been merged into librdkafka_vendor.

See https://github.com/edenhill/librdkafka/pull/4033

### CODE REVIEW

**Review by edenhill — CHANGES_REQUESTED** [review]

File: kafka/consumer.go [hunk 1]
```diff
@@ -907,3 +907,8 @@ func (c *Consumer) handleRebalanceEvent(channel chan Event, rkev *C.rd_kafka_eve
 
 	return nil
 }
+
+// SaslSetCredentials (re)sets the SASL credentials on this consumer.
+func (c *Consumer) SaslSetCredentials(username, password string) error {
```
[the remark this answers is not in the document]
Comment (josvisser66) [reply kafka/consumer.go:912]: Done

File: kafka/kafka.go [hunk 2]
```diff
@@ -373,3 +374,22 @@ func LibraryVersion() (int, string) {
...
+	err := C.rd_kafka_sasl_set_credentials(rk, cUsername, cPassword)
+	if err == nil {
+		return nil
+	}
+
+	defer C.free(unsafe.Pointer(err))
+	cErrMsg := C.rd_kafka_error_string(err)
+	defer C.free(unsafe.Pointer(cErrMsg))
+
+	return errors.New(C.GoString(cErrMsg))
```
[the remark this answers is not in the document]
Comment (josvisser66) [reply kafka/kafka.go:394]: Done

File: kafka/consumer.go [hunk 3]
```diff
@@ -907,3 +907,8 @@ func (c *Consumer) handleRebalanceEvent(channel chan Event, rkev *C.rd_kafka_eve
 
 	return nil
 }
+
+// SaslSetCredentials (re)sets the SASL credentials on this consumer.
```
[the remark this answers is not in the document]
Comment (josvisser66) [reply kafka/consumer.go:911]: Fixed comments to make them more descriptive.

File: kafka/consumer.go [hunk 4]
```diff
@@ -907,3 +907,13 @@ func (c *Consumer) handleRebalanceEvent(channel chan Event, rkev *C.rd_kafka_eve
 
 	return nil
 }
+
+// SetSaslCredentials sets the SASL credentials used for this consumer. The new credentials
+// will overwrite the old ones (which were set when creating the consumer or by a previous
+// call to SetSaslCredentials). The new credentials will be used the next time the
+// consumer needs to establish a connection to the broker. This method will *not*
```
Comment (edenhill) [inline kafka/consumer.go:914]: I think "needs to authenticate to brokers" is more correct, and will also cover the future case of reauthentication

**Review by edenhill — CHANGES_REQUESTED** [review]
some minor docstring comments.
This method also needs to be added to the admin client in admin.go

Please also add a line to the CHANGELOG

File: kafka/consumer.go (hunk 4 above)
Comment (josvisser66) [reply kafka/consumer.go]: Both done

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # Confluent's Golang client for Apache Kafka
 
+## v1.9.3
+
+This is a maintenaince release:
+
+ * Added SetSaslCredentials wrapper for librdkafka rd_kafka_sasl_set_credentials.
```
Comment (edenhill) [inline CHANGELOG.md:7]: The librdkafka function name is not relevant to users of the Go client - it is merely an implementation detail - instead describe what the new method does/allows

**Review by edenhill — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (josvisser66) [reply CHANGELOG.md]: Done.

**Comment by edenhill** [comment]
close-reopen tro trigger CI

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,13 @@
 # Confluent's Golang client for Apache Kafka
 
+## v1.9.3
+
+This is a maintenaince release:
+
+ * Added SetSaslCredentials. This new method (on the Producer, Consumer, and
+   AdminClient) allows modifying the stored SASL PLAIN/SCRAM credentials that
+	 will be used for subsequent (new) connections to a broker.
```
Comment (edenhill) [inline CHANGELOG.md:9]: nit: weird indent, align with previous line.

**Review by edenhill — CHANGES_REQUESTED** [review]
We would need at least a dry test for this, just to make sure the API doesn't crash.
I recommend adding a call to the end of TestProducerAPIs() with a call to SetSaslCredentials():
https://github.com/confluentinc/confluent-kafka-go/blob/master/kafka/producer_test.go#L182

File: CHANGELOG.md (hunk 6 above)
Comment (josvisser66) [reply CHANGELOG.md]: Done and done

**Review by edenhill — APPROVED** [review]
LGTM, thanks Jos!

**Pull request merged**